src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -183,8 +189,53 @@ object DatabricksPlugin extends AutoPlugin {
     }
   }
 
+  private lazy val executeCommandImpl: Def.Initialize[Task[Seq[CommandStatus]]] = Def.taskDyn {
+    Def.task {
+      val client = dbcApiClient.value
+      val language = dbcExecutionLanguage.value
+      val onClusters = dbcClusters.value
+      val (allClusters, _) = dbcFetchClusters.value
+      val commandFile = dbcCommandFile.value
+      val commandStatuses = Seq.empty[CommandStatus]
+      client.foreachCluster(onClusters, allClusters) { confirmedCluster =>
+        var contextCorrect = false
+        val contextId = client.createContext(language, confirmedCluster)
+        while (!contextCorrect) {
+          val contextStatus = client.checkContext(contextId, confirmedCluster)
+          if (contextStatus.status == DBCContextRunning.status) {
+            contextCorrect = true
+          } else if (contextStatus.status == DBCContextError.status) {
+            client.destroyContext(contextId, confirmedCluster)
+            contextCorrect = true
+            Def.task(throw new RuntimeException(s"The Execution Context '${contextId.id} returned an error. Terminated context"))
+          }
+        }
+
+        // Attempt to issue a command
+        val commandId = client.executeCommand(language, confirmedCluster, contextId, commandFile)
+        var commandComplete = false
+        while (!commandComplete) {
+          val commandStatus = client.checkCommand(confirmedCluster, contextId, commandId)
+          if (commandStatus.status == DBCCommandFinished.status) {
+            commandStatuses :+ commandStatus
+            commandComplete = true
+          } else if (commandStatus.status == DBCCommandError.status) {
+            client.cancelCommand(confirmedCluster, contextId, commandId)
+            client.destroyContext(contextId, confirmedCluster)
+            commandComplete = true
+            Def.task(throw new RuntimeException(s"The command '${commandId.id} returned an error. Cancelled command and terminated context"))
+          } else {
+            Thread sleep 3000
+          }
+        }
+        client.destroyContext(contextId, confirmedCluster)
+      }
+      commandStatuses
+    }
+  }
+
   val baseDBCSettings: Seq[Setting[_]] = Seq(
-    dbcUsername := { 
+    dbcUsername := {
```

Comment:
@brkyvz just pushed these changes and I was then able to get rid of taskDyn (cheers again). 

The problem still exists however of double execution of the `destroyClient` calls in test no.15. it appears that the `Def.task(throw RuntimeException...` masks the exception and it does not exit the loop.